src/integrations/Elastic.Apm.AspNetFullFramework/Extensions/SoapRequest.cs

```diff
@@ -102,20 +104,22 @@ internal static string GetSoap12ActionFromHttpRequest(IApmLogger logger, HttpReq
 
 		internal static string GetSoap12ActionFromInputStream(IApmLogger logger, Stream stream)
 		{
-			var shared = ArrayPool<byte>.Shared;
-			var bytes = shared.Rent((int)stream.Length);
+			if (stream.Length > MaxRequestBodySizeInBytesToCapture)
+			{
+				logger.Info()
+					?.Log(
+						$"SOAP request body exceeds {MaxRequestBodySizeInBytesToCapture} bytes - skipping request body capturing to limit memory usage");
+				return null;
```

Comment:
Is there any value in capturing the initial bytes, as much as we can fit in the buffer? If not, should we skip renting the array at all?

Reply:
Yeah, that makes sense. I adapted the code to always capture up to `MaxRequestBodySizeInBytesToCapture`.